noteable/sql/sqlalchemy/utils.py

```diff
@@ -0,0 +1,238 @@
+            underlying_schemas.append(default_schema)
+
+        # Honor avoiding schemas_to_avoid case insensitively
+        return [s for s in underlying_schemas if s.lower() not in self.schemas_to_avoid]
+
+    def get_columns(self, relation_name: str, schema: Optional[str] = None) -> List[dict]:
+        """Call the underlying get_columns(), but convert the type object members
+        to strings, not SQLA-centric objects."""
+        columns: List[dict] = self.underlying_inspector.get_columns(relation_name, schema=schema)
+
+        for col in columns:
+            col['type'] = _determine_column_type_name(col['type'])
+
+            # Some dialects do not return at all, but are expected to.
+            if 'comment' not in col:
+                col['comment'] = ''
+
+        return columns
+
+    @handle_not_implemented('(unobtainable)')
+    def get_view_definition(self, view_name: str, schema: Optional[str] = None) -> str:
+        return self.underlying_inspector.get_view_definition(view_name, schema=schema)
+
+    def get_pk_constraint(self, table_name: str, schema: Optional[str] = None) -> dict:
+        return self.underlying_inspector.get_pk_constraint(table_name, schema=schema)
+
+    def get_foreign_keys(self, table_name: str, schema: Optional[str] = None) -> List[dict]:
+        return self.underlying_inspector.get_foreign_keys(table_name, schema=schema)
+
+    @handle_not_implemented(default_factory=list)
+    def get_check_constraints(self, table_name: str, schema: Optional[str] = None) -> List[dict]:
+        return self.underlying_inspector.get_check_constraints(table_name, schema=schema)
+
+    def get_indexes(self, table_name: str, schema: Optional[str] = None) -> List[dict]:
+        return self.underlying_inspector.get_indexes(table_name, schema=schema)
+
+    @handle_not_implemented(default_factory=list)
+    def get_unique_constraints(self, table_name: str, schema: Optional[str] = None) -> List[dict]:
+        return self.underlying_inspector.get_unique_constraints(table_name, schema=schema)
+
+    def get_table_names(self, schema: Optional[str] = None) -> List[str]:
+        return self.underlying_inspector.get_table_names(schema)
+
+    def get_view_names(self, schema: Optional[str] = None) -> List[str]:
+        return self.underlying_inspector.get_view_names(schema)
+
+
+class AthenaInspector(WrappedInspector):
+    def get_pk_constraint(self, table_name: str, schema: Optional[str] = None) -> Optional[dict]:
+        # Athena does not support PKs, and the underlying dialect returns emtpy _list_,
+        # not an empty _dict_!
+
+        # Athena dialect returns ... an empty _list_ instead of a dict, contrary to what
+        # https://docs.sqlalchemy.org/en/14/core/reflection.html#sqlalchemy.engine.reflection.Inspector.get_pk_constraint
+        # specifies for the return result from inspector.get_pk_constraint().
+
+        return None
+
+
+class BigQueryInspector(WrappedInspector):
```

Comment:
The rest of `SchemaStrippingInspector`, but only needed to kick in for BigQuery.